File: TAF.Core/Car/CarInfo.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="CarInfo.cs" company="" author="何翔华">
//   
// </copyright>
// <summary>
//   CarInfo
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace SCBF.Car
{
    using SCBF.BaseInfo;
    using System;
    using System.Collections.Generic;

    /// <summary>
    /// 车辆信息
    /// </summary>
    public class CarInfo : TAFEntity
    {
        /// <summary>
        /// 车辆型号
        /// </summary>
        public string Clxh { get; set; }

        /// <summary>
        /// 车架号
        /// </summary>
        public string Cjh { get; set; }

        /// <summary>
        /// 发动机号
        /// </summary>
        public string Fdjh { get; set; }

        /// <summary>
        /// 油料标号
        /// </summary>
        public Guid? OctaneRatingId { get; set; }

        /// <summary>
        /// 车牌号
        /// </summary>
        public string Cph { get; set; }

        /// <summary>
        /// 基础公里数
        /// </summary>
        public decimal Jcgls { get; set; }

        /// <summary>
        /// 装备时间
        /// </summary>
        public DateTime Zbsj { get; set; }

        /// <summary>
        /// 整备资料
        /// </summary>
        public string Zbzl { get; set; }

        /// <summary>
        /// 行驶证号
        /// </summary>
        public string Xszh { get; set; }

        /// <summary>
        /// 油箱限额
        /// </summary>
        public decimal Yxxe { get; set; }

        /// <summary>
        /// 车辆状况
        /// </summary>
        public Guid ClzkId { get; set; }

        /// <summary>
        /// 车辆状况
        /// </summary>
        public virtual SysDictionary Clzk { get; set; }

        /// <summary>
        /// 驾驶员Id
        /// </summary>
        public Guid? DriverId { get; set; }

        /// <summary>
        /// 夏季油耗
        /// </summary>
        public decimal? OilWearSummer { get; set; }

        /// <summary>
        /// 冬季油耗
        /// </summary>
        public decimal? OilWearWinter { get; set; }

        /// <summary>
        /// 驾驶员
        /// </summary>
        public virtual Driver Driver { get; set; }

        /// <summary>
        /// 油料卡
        /// </summary>
        public virtual List<OilCard> OilCards { get; set; }

        /// <summary>
        /// 实物油料加油记录
        /// </summary>
        public virtual List<ApplicationForBunkerB> ApplicationForBunkerBs { get; set; }

        /// <summary>
        /// 车辆维修记录
        /// </summary>
        public virtual List<ApplyForVehicleMaintenance> ApplyForVehicleMaintenances { get; set; }

    }
}

File: TAF.Web/Views/CarInfo/carInfo.js
Vue.component("form-body", {
    mixins: [itemMixin],
    template: "#carInfoFormBody",
    ready: function () {
        var $this = this;
        var datePickerZbsj = $('#datePickerZbsj').datepicker({ format: 'yyyy-mm-dd' }).on('changeDate', function (ev) {
            datePickerZbsj.hide();
            $this.item.zbsj = $("#datePickerZbsj").val();
        })
        .on('hide', function (event) {
		        event.preventDefault();
		        event.stopPropagation();
	    })
            .data('datepicker');

        $("#clzk").select2().on("change", function (e) { $this.item.clzk = $("#clzk").val(); });
        $("#driver").select2().on("change", function (e) { $this.item.driver = $("#driver").val(); });
        $("#octaneRatingId").select2().on("change", function (e) { $this.item.octaneRatingId = $("#octaneRatingId").val(); });
    },
    data: function () {
        return {
            item: {
                id: "",            
                clxh: "",
                cjh: "",
                fdjh: "",
                octaneRatingId: "",
                cph: "",
                jcgls: 0,
                zbsj: "",
                clzk: "",
                xszh: "",
                yxxe: 0,
                zbzl: "",
                driver: "",
                oilWearSummer: 0,
                oilWearWinter:0
            }
        };
    },
    events: {
        'onSaveItem': function (closeModal) {
            var $this = this;
            abp.services.app.carInfo.saveAsync($this.item)
            .done(function (m) {
                $this.done(closeModal);
            })
            .fail(function (m) {
                $this.fail(m);
            });
        },
        'onGetItem': function (id) {
            this.editModel = true;
            var $this = this;
            abp.services.app.carInfo.get(id)
                .done(function (m) {
                    $this.item = m;
                    $("#clzk").select2().val(m.clzk).trigger("change");
                    $("#driver").select2().val(m.driver).trigger("change");
                    $("#octaneRatingId").select2().val(m.octaneRatingId).trigger("change");
                })
            .fail(function (m) {
                $this.fail(m);
            });
        }
    },
    methods: {
        clearItem: function () {
            this.item.id = "";
            this.item.clxh= "";
            this.item.cjh= "";
            this.item.fdjh= "";
            this.item.octaneRatingId= "";
            this.item.cph= "";
            this.item.jcgls= 0;
            this.item.zbsj= "";
            this.item.clzk= "";
            this.item.xszh= "";
            this.item.zbzl= "";
            this.item.driver= "";
            this.item.yxxe= 0;
            this.item.oilWearSummer= 0;
            this.item.oilWearWinter = 0;
            $("#clzk").select2().val("").trigger("change");
            $("#driver").select2().val("").trigger("change");
            $("#octaneRatingId").select2().val("").trigger("change");
        }
    }
});


var main = new Vue({
    mixins: [indexMixin],
    ready: function () {
        $("#searchClzk").select2().on("change", function (e) { main.queryEntity.clzk = $("#searchClzk").val(); });
        $("#searchDriver").select2().on("change", function (e) { main.queryEntity.driver = $("#searchDriver").val(); });
    },
    data: {
        queryEntity: {
            clxh:"", 
            cjh:"", 
            fdjh:"", 
            cph:"", 
            zbsjFrom:"", 
            zbsjTo:"", 
            clzk:"", 
            xszh: "",
            driver:""
        },
        list4:[],
        selectId:"",
        modifyEntity: { modifyTitle: "" }
    },
    events: {
        'onResetSearch': function () {
            this.queryEntity.clxh="";
            this.queryEntity.driver="";
            this.queryEntity.cjh="";
            this.queryEntity.fdjh="";
            this.queryEntity.cph="";
            this.queryEntity.zbsjFrom=""; 
            this.queryEntity.zbsjTo="";
            this.queryEntity.clzk="";
            this.queryEntity.xszh = "";
            $("#searchClzk").select2().val("").trigger("change");
            $("#searchDriver").select2().val("").trigger("change");
        }
    },
    methods: {
        excuteQuery: function ($this) {
            abp.services.app.carInfo.getAll(main.queryEntity)
                .done(function (r) {
                    $this.bindItems($this, r);
                });
        },
        delete: function (id) {
            var $this = this;
            abp.services.app.carInfo.delete(id)
                .done(function (m) {
                    $this.done();
                })
                .fail(function (m) {
                    $this.fail(m);
                });
        },
        showAttachmentsDialog: function(name, id) {
            var $this = this;
            $this.modifyEntity.modifyTitle = name;
            $this.selectId = id;
            $("#attachmentsDialog").modal("show");
            $this.query4();

        },
        removeAttach: function(id) {
            var $this = this;
            abp.services.app.attachment.delete(id)
                .done(function(m) {
                    $this.query4();
                })
                .fail(function(m) {
                    $this.fail(m);
                });
        },
        query4: function() {
            var $this = this;

            abp.services.app.attachment.getAll($this.selectId)
                .done(function(r) {
                    $this.list4 = r;
                })
                .fail(function(r) {
                    $this.fail(r);
                });
        },
    }
});

main.query(0);


$(".fileUpload").liteUploader({
        script: defaultUrl + "ProcessManagement/Upload?modelId=" + main.selectId
    })
    .on("lu:success",
        function(e, response) {
            main.query4();
            taf.notify.success("附件上传成功");
        });

$(".fileUpload").change(function() {
    $(".fileUpload").data("liteUploader").options.script =
        defaultUrl + "ProcessManagement/Upload?modelId=" + main.selectId;
    $(this).data("liteUploader").startUpload();
});

